crates/biome_html_formatter/src/utils/metadata.rs

```diff
@@ -716,8 +716,19 @@ pub(crate) fn is_canonical_html_tag_name(tag_name: &str) -> bool {
     }
 }
 
-/// Whether the given tag name is a known HTML element. See also: [`HTML_ALL_TAGS`].
-pub(crate) fn is_canonical_html_tag(tag_name: &HtmlTagName) -> bool {
+/// Determines if a tag name should be lowercased as a canonical HTML tag.
+///
+/// Only returns true for pure HTML files (.html). For component framework files
+/// (Svelte, Astro, Vue), this always returns false to preserve component name casing.
+///
+/// This ensures that component names like `<Button>` are not lowercased to `<button>`
+/// in component files, while maintaining the lowercasing behavior for HTML files.
```

Comment:
nit: This doc comment is a regression because we loose the link to `HTML_ALL_TAGS`.